benchmarks/scripts/expval_benchmark.py

```diff
@@ -158,35 +159,91 @@ def fetch_pre_post_compiled_circuits(
     return uncompiled_qiskit_circuit, compiled_qiskit_circuit
 
 
+def get_heavy_bitstrings(circuit: qiskit.QuantumCircuit) -> Set[str]:
+    """ "Determine the heavy bitstrings of the circuit."""
+    simulator = AerSimulator(method="statevector")
+    result = simulator.run(circuit).result()
+    counts = list(result.get_counts().items())
+    median = np.median([c for (_, c) in counts])
+    return set(bitstring for (bitstring, p) in counts if p > median)
+
+
+def estimate_heavy_output(
+    circuit: qiskit.QuantumCircuit,
+    qv_1q_err: float = 0.002,
+    qv_2q_err: float = 0.02,
+) -> float:
+    """Sample the heavy bitstrings on the backend and estimate the heavy output
+    probability from the counts of the heavy bitstrings.
+
+    Args:
+        circuit: The circuit for which to compute the heavy output metric.
+        qv_1q_err: The single qubit error rate for the backend noise model.
+        qv_2q_err: The two-qubit error rate for the backend noise model.
+
+    Returns:
+        The heavy output probability as a float.
+    """
+    heavy_bitstrings = get_heavy_bitstrings(circuit)
+
+    simulator = AerSimulator(
+        method="statevector",
+        noise_model=create_depolarizing_noise_model(
+            circuit, qv_1q_err, qv_2q_err
+        ),
+    )
+    result = simulator.run(circuit).result()
+
+    heavy_counts = sum(
+        [
+            result.get_counts().get(bitstring, 0)
+            for bitstring in heavy_bitstrings
+        ]
+    )
+    nshots = 10000
+    hop = (
+        heavy_counts - 2 * math.sqrt(heavy_counts * (nshots - heavy_counts))
+    ) / nshots
+    return hop
+
+
 def simulate_expvals(
     uncompiled_circuit: qiskit.QuantumCircuit,
     compiled_circuit: qiskit.QuantumCircuit,
-    observable: str,
+    circuit_name: str,
 ) -> tuple[float, float]:
     """Simulates the expectation values of a given observable for
     both uncompiled and compiled quantum circuits.
 
     Args:
         uncompiled_circuit: The original quantum circuit before compilation.
         compiled_circuit: The quantum circuit after compilation.
-        observable: The observable for which the expectation value
-            is to be calculated, represented as a string.
+        circuit_name: The name of the quantum circuit in string format.
 
     Returns:
         A tuple containing the expectation values of the observable for the
         uncompiled and compiled circuits, respectively.
     """
+    if circuit_name == "qv":
+        compiled_circuit.measure_all()
+        uncompiled_circuit.measure_all()
+        return (
+            estimate_heavy_output(compiled_circuit),
+            estimate_heavy_output(uncompiled_circuit, 0, 0),
+            "HOP",
```

Comment:
Instead of having two arguments for noise rates, can we have a flag for noisy/ideal simulation? If noisy simulation, then use the global variables.

Reply:
My rule of thumb is explicit over implicit. I'd consider either @natestemen 's suggestion, or just explicitly passing the noise rates in both calls.

Reply:
agreed, used Nate's suggestion, except I did customize the error rates for QV.

Reply:
What's the rationale for having different error rates for QV?

Reply:
That makes it sound like the rationale to have different error-rates is to portray UCC as the best, which doesn't necessarily feel like a fair comparison. I think we need a stronger reason to use different error rates. I'm sure there are good ones, I just don't know them.

If we do keep different error-rates, then I think it makes sense to record that data in the results files.

Reply:
In light of this discussion, I will open a new PR to change the error rates back to the global variables.